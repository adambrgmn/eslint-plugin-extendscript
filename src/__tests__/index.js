import { CLIEngine } from 'eslint';
import plugin from '../index';

const createCli = env => {
  const cli = new CLIEngine({
    useEslintrc: false,
    baseConfig: {
      extends: 'eslint:recommended',
      plugins: ['extendscript'],
      env,
    },
  });
  cli.addPlugin('eslint-plugin-extendscript', plugin);
  return cli;
};

const lint = (code, cli) => {
  const linter = cli.executeOnText(code);
  return CLIEngine.getErrorResults(linter.results);
};

test('Separate environments', () => {
  const cli = createCli({
    'extendscript/base': true,
    'extendscript/indesign': true,
    'extendscript/illustrator': true,
    'extendscript/photoshop': true,
    'extendscript/scriptui': true,
  });

  const result = lint(
    `
    $.writeln('Hello world!');
    new AlignOptions();
    new Artboard();
    new Channel();
    new ScriptUI();
  `,
    cli,
  );

  expect(result).toMatchSnapshot();
});

test('General environment', () => {
  const cli = createCli({ 'extendscript/extendscript': true });

  const result = lint(
    `
    $.writeln('Hello world!');
    new Book();
    new BlendModes();
    new Direction();
    new DropDownList();
  `,
    cli,
  );

  expect(result).toMatchSnapshot();
});

test('Should not leak between environments', () => {
  const cli = createCli({ 'extendscript/indesign': true });
  const result = lint(
    `
    $.writeln('Hello world');
  `,
    cli,
  );

  expect(result).toMatchSnapshot();
});
