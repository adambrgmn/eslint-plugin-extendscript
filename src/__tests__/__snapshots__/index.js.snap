// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General environment 1`] = `Array []`;

exports[`Separate environments 1`] = `Array []`;

exports[`Should not leak between environments 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "<text>",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 5,
        "endColumn": 6,
        "endLine": 2,
        "line": 2,
        "message": "'$' is not defined.",
        "nodeType": "Identifier",
        "ruleId": "no-undef",
        "severity": 2,
        "source": "    $.writeln('Hello world');",
      },
    ],
    "source": "
    $.writeln('Hello world');
  ",
    "warningCount": 0,
  },
]
`;
